### PR TITLE
Adds PHP 8.3 as new supported version

### PR DESCRIPTION
**Pull request by fezfez**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,13 +10,13 @@ to provide a minimalist PSR-7 middleware framework for PHP, with the following
...
+  - [Aura.Router](https://github.com/auraphp/Aura.Router)
     - [FastRoute](https://github.com/nikic/FastRoute)
     - [laminas-router](https://github.com/mezzio/mezzio-router)
 - DI Containers, via [PSR-11 Container](https://github.com/php-fig/container).
   Middleware matched via routing is retrieved from the composed container.
 - Optionally, templating. We support:
-    - [Plates](http://platesphp.com/)
+  - [Plates](http://platesphp.com/)
     - [Twig](http://twig.sensiolabs.org/)
     - [Laminas's PhpRenderer](https://github.com/laminas/laminas-view)
```
Comment (gsteel) [inline README.md]: These lists are all nested weird now - can you reduce the indent on all of the list items please?

File: psalm-baseline.xml [hunk 2]
```diff
@@ -176,6 +179,9 @@
     </PropertyNotSetInConstructor>
   </file>
   <file src="test/ConfigProviderTest.php">
+    <DeprecatedClass>
+      <code>new Config($dependencies)</code>
+    </DeprecatedClass>
```
Comment (gsteel) [inline psalm-baseline.xml]: This deprecation is pretty easy to deal with, with SM at ^3, we can just pass a regular array to the SM constructor and remove the usage of the `Config` class in the test. Could you take a look at that whilst you're here please?

See https://github.com/mezzio/mezzio/blob/2bac6f637837fce15239a358b7e13a71c400d5c4/test/ConfigProviderTest.php#L139-L146

Pretty sure this method can just be `return new ServiceManager($dependencies)`

Also, the `psalm-type` is outdated here - it should probably be `@psalm-import-type ServiceManagerConfiguration from ServiceManager`

**Review by gsteel — CHANGES_REQUESTED** [review]
Thanks @fezfez - Just a couple of tweaks :)

File: README.md (hunk 1 above)
Comment (fezfez) [reply README.md]: done

File: psalm-baseline.xml (hunk 2 above)
Comment (fezfez) [reply psalm-baseline.xml]: done

**Review by gsteel — APPROVED** [review]
Thanks @fezfez

**Pull request merged**